src/kOS.Safe/Compilation/KS/KSScript.cs

```diff
@@ -97,35 +97,79 @@ public override bool IsCommandComplete(string command)
-            int openCurlyBrackets = 0;
-            int openParentheses = 0;
-
+            bool inQuotes = false;
+            bool inCommentToEoln = false;
+            bool waitForMoreTokens = false;
+            char curChar;
+            char prevChar = '\0';
+
+            // First, we have to check manually for unterminated string literals because
+            // they are a continuation in the midst of a token, rather than between tokens,
+            // and thus the parser doesn't quite catch them the same way.
             for (int n = 0; n < length; n++)
             {
-                switch (commandChars[n])
+                curChar = commandChars[n];
+                switch (curChar)
                 {
-                    // match curly brackets
-                    case '{':
-                        openCurlyBrackets++;
+                    // Track if we are ina string literal that didn't close,
+                    // and make sure it's not a string literal inside a comment,
+                    // becasue those don't count:
+                    case '\"':
+                        if (! inCommentToEoln)
+                            inQuotes = !(inQuotes);
                         break;
-
-                    case '}':
-                        openCurlyBrackets--;
+                    case '/':
+                        if (prevChar == '/' && !inQuotes )
+                            inCommentToEoln = true;
                         break;
-                    // match parentheses
-                    case '(':
-                        openParentheses++;
+                    case '\n':
+                    case '\r':
+                        inCommentToEoln = false;
                         break;
-
-                    case ')':
-                        openParentheses--;
+                }
+                prevChar = curChar;
+            }
+            
+            // Second, if we aren't in an unterminated literal string, then let
+            // the parser do the rest of the checking by seeing if it reports
+            //    Unexpected Token 'EOF', and looking at what it was expecting instead.
+
+            // - Possible future refactor - 
+            // The string comparison of the human-readable message is the only way
+            // to find out if the error is the exact one we're looking for, which
+            // is what the code below does, and that's a bit fragile.
+            // Making it use a more robust check would first require editing the
+            // TinyPG C# source code and changing the way it encodes a ParseError
+            // so it stores that sort of thing as separate pieces of data in its members.
+            
+            if (!inQuotes)
+            {
+                ParseTree parseTree = parser.Parse(command);
+                
+                foreach (ParseError err in parseTree.Errors)
+                {
+                    System.Console.WriteLine("eraseme:  err msg: " + err.Message);
```

Comment:
I should make a small bash script to wrap around my 'git push' es  that greps for my 'eraseme' string throughout all the diffs and warns me if there are any present, asking for confirmation before actually pushing the change.